rustls/src/crypto/aws_lc_rs/mod.rs

```diff
@@ -35,23 +36,29 @@ pub(crate) mod tls13;
 /// A `CryptoProvider` backed by aws-lc-rs.
 pub fn default_provider() -> CryptoProvider {
     CryptoProvider {
-        // TODO: make this filtering conditional on fips feature
-        cipher_suites: DEFAULT_CIPHER_SUITES
-            .iter()
-            .filter(|cs| cs.fips_mode())
-            .copied()
-            .collect(),
-        kx_groups: ALL_KX_GROUPS
-            .iter()
-            .filter(|kx| kx.fips_mode())
-            .copied()
-            .collect(),
+        cipher_suites: DEFAULT_CIPHER_SUITES.to_vec(),
```

Comment:
Hmm, how about we add a `default_fips_provider()` which is only available on `feature("fips")`?